src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -1006,7 +1006,7 @@ private bool TryOpen(TaskCompletionSource<DbConnectionInternal> retry)
             }
 
             if (StatisticsEnabled ||
-                s_diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
+                (s_diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand) && Statistics != null ))
```

Comment:
Separately, what causes _statistics to be set to null or non-null?  Is it possible it could transition from non-null to null concurrently with this code executing?  If so, it should be stored into a local and then that local checked.